library/validator.py

```diff
@@ -0,0 +1,84 @@
+import yaml
+
+class Validator:
+    """ 
+    Validator takes as input the path of a configuration file 
+    and will run the necessary checks to determine wether the structure
+    and values of the files are valid according to the requirements of
+    the library.
+    """
+
+    def __init__(self):               
+        # Correct fields, children and data types
+        # 'field_name': ['d_type': class, 'nullable': boolean']
+        self.valid_tree = {
+            'name': {'d_type': str, 'nullable': False},
+            'version': {'d_type': str, 'nullable': False},
+            'acl': {'d_type': str, 'nullable': False},
+            
+            'source': {
+                # add url or socrata
+                'options': {'d_type': [], 'nullable': False},
+                'geometry': {
+                    'SRS': {'d_type': str, 'nullable': False},
+                    'type': {'d_type': str, 'nullable': False}
+                }
+            },
+
+            'destination': {
+                'name': {'d_type': str, 'nullable': False},
+                'geometry': {
+                    'SRS': {'d_type': str, 'nullable': False},
+                    'type': {'d_type': str, 'nullable': False}
+                },
+                'options': {'d_type': [], 'nullable': False},
+                'fields': {'d_type': [], 'nullable': False},
+                'sql': {'d_type': str, 'nullable': True}  
+            },
+
+            # Some fields are optional
+            'info': {
+                'description': {'d_type': str, 'nullable': True},
+                'url': {'d_type': str, 'nullable': True},
+                'dependent': {'d_type': str, 'nullable': True}
+            }
+        }
+
+    def __load_file(self, path):
```

Comment:
this can be a cached property, also make "path" a class attribute under def __init__

Reply:
e.g. 
```python
@cached_property
def _file(self):
    .....
```

so that for the specific validators, you can do something like:

```python
@property
def dataset_name_matches(self) -> bool:
      dataset = self._file['dataset']
      return (dataset['name'] == name) and (dataset['name'] == dataset['destination']['name'])
```

and in your test, instead of passing the file as a paramater,
you can just test on these things as class properties
```python
v = Validator()
print(v.dataset_name_matches)
```

Reply:
Also the whole validation process should be wrapped in a `def __call__(self)` method